src/index.ts

```diff
@@ -8,33 +8,168 @@ export interface SnowFlake {
     id: number;
 }
 
-/*
- * The read message interface
+/**
+ * Sent when a message is create/updated. The inner payload is a message object.
  */
-export interface ReadMessage {
+export interface MessageEvent {
+    /**
+     * id of the message
+     */
+    id: SnowFlake;
 
     /**
-     * The message contents (up to 2000 characters)
+     * id of the channel the message was sent in
      */
-    content: string;
+    channel_id: SnowFlake;
+
+    /**
+     * id of the guild the message was sent in
+     */
+    guild_id?: SnowFlake;
+
+    /**
+     * user object
+     */
+    author: User;
 
     /**
-     * A nonce that can be used for optimistic message sending
+     * contents of the message
      */
-    nonce: SnowFlake;
+    content: string;
 
     /**
-     * true if this is a TTS message.
+     * whether this was a TTS message
      */
     tts: boolean;
 
     /**
-     * The contents of the file being sent
+     * any attached files
+     */
+    attachments: Attachment[];
+
+    /**
+     * used for validating a message was sent
+     */
+    nonce?: SnowFlake;
```

Comment:
Instead of writing `?` which it means that it's not required for the interface but actually it's. So, replace it with `undefined`.

For Example: 
```ts
nonce: SnowFlake | undefined;
```